### PR TITLE
[RpcServer] Update RpcServer.SmartContract.cs

### PR DESCRIPTION
**Pull request by chenzhitong**
close https://github.com/neo-project/neo-modules/issues/398

### CODE REVIEW

**Review by shargon — CHANGES_REQUESTED** [review]
I think that this properties can't be null or compute the `Hash` will fault.

**Comment by chenzhitong** [comment]
> I think that this properties can't be null or compute the `Hash` will fault.

So we should make `signers` a normal parameter instead of an optional parameter?
``` c#
GetInvokeResult(byte[] script, Signers signers = null)
```

**Comment by superboyiii** [comment]
> I think that this properties can't be null or compute the `Hash` will fault.

It could be null. If it's null, just do nothing in `ProcessInvokeWithWallet`.
https://github.com/neo-project/neo-modules/blob/2ae6a9722b7a6a031943f92866d53c89b18b3433/src/RpcServer/RpcServer.Wallet.cs#L168

**Comment by cloud8little** [comment]
I think default  signers value should not be null when calling invokefunction and invokescript. It should be Array.Empty<Signer>().

https://github.com/neo-project/neo-modules/blob/2ae6a9722b7a6a031943f92866d53c89b18b3433/src/RpcServer/RpcServer.SmartContract.cs#L130

https://github.com/neo-project/neo-modules/blob/2ae6a9722b7a6a031943f92866d53c89b18b3433/src/RpcServer/RpcServer.SmartContract.cs#L116

**Comment by shargon** [comment]
If `Hash` FAULT, It can FAULT here https://github.com/neo-project/neo/blob/2d0e6123a1f9e58c8069f0762d17c6e5eec2dde9/src/neo/Network/P2P/Payloads/Transaction.cs#L325

**Comment by superboyiii** [comment]
> If `Hash` FAULT, It can FAULT here https://github.com/neo-project/neo/blob/2d0e6123a1f9e58c8069f0762d17c6e5eec2dde9/src/neo/Network/P2P/Payloads/Transaction.cs#L325

Hash will not be computed if `signer` is null. 
https://github.com/neo-project/neo-modules/blob/f13d7c305f02f77aa48cc579f678a96e4f319a25/src/RpcServer/RpcServer.Wallet.cs#L171

**Comment by shargon** [comment]
Could you test it calling `InvokeFunction` to a method that return the hash of the current transaction?
https://github.com/neo-project/neo-modules/blob/8f03a7f78fa0de02385574b4aedc28b21c0d81c3/src/RpcServer/RpcServer.SmartContract.cs#L116

**Comment by superboyiii** [comment]
> Could you test it calling `InvokeFunction` to a method that return the hash of the current transaction (without signer)?
> https://github.com/neo-project/neo-modules/blob/8f03a7f78fa0de02385574b4aedc28b21c0d81c3/src/RpcServer/RpcServer.SmartContract.cs#L116

Do you mean return hash in the internal of SC? If in this situation, it will truely return fault.

**Comment by Tommo-L** [comment]
In some cases, we don't need to set `Signer`, like invoke `balanceOf` method.  And if the invoke return faults, the user need to check it.

File: src/RpcServer/RpcServer.SmartContract.cs [hunk 1]
```diff
@@ -65,9 +65,9 @@ private JObject GetInvokeResult(byte[] script, Signers signers = null)
         {
             Transaction tx = new Transaction
```
Comment (shargon) [inline src/RpcServer/RpcServer.SmartContract.cs:66]:
```suggestion
            Transaction tx = signers == null ? null : new Transaction
```

**Review by shargon — APPROVED** [review]

**Pull request merged**